de.metas.swat/de.metas.swat.base/src/main/java/de/metas/adempiere/form/terminal/AbstractTerminalNumericField.java

```diff
@@ -327,34 +327,52 @@ else if (value instanceof String)
 	@Override
 	protected final void setFieldValue(final BigDecimal value, final boolean fireEvent)
 	{
+		final BigDecimal valueOld = this._valueOld; // backup for event
+		
 		//
 		// Fix value to set
-		final BigDecimal valueToSet;
+		final BigDecimal valueNew;
 		if (value == null)
 		{
-			valueToSet = Env.ZERO;
+			valueNew = Env.ZERO;
 		}
 		else
 		{
 			// Qty Editor it shall be a small component and we don't want to clutter it with pointless trailing zeros (06112)
-			valueToSet = NumberUtils.stripTrailingDecimalZeros(value);
+			valueNew = NumberUtils.stripTrailingDecimalZeros(value);
 		}
 
-		final BigDecimal valueOld2 = valueOld; // backup for event
-		final BigDecimal valueNew2 = valueToSet; // backup for event
-		final boolean changed = valueOld != null && valueOld.compareTo(valueToSet) != 0;
+		//
+		// Get the number editor
+		final ITerminalTextField fNumber = this.fNumber;
+		if (isDisposed() || fNumber == null)
```

Comment:
the TerminalException is also thorwn if isDisposed()==true...so the TErminalException might pop up more often..
assuming each time they had to click twice, then now have to click awoild the terminalexception now..
wdyt about: if isDisposed(), but fNumber != null, and we are not in dev-mode, then only log the exception, but not throw it?